appscale/tools/remote_helper.py

```diff
@@ -131,14 +131,30 @@ def start_all_nodes(cls, options, node_layout):
 
     if login_ip in public_ips:
       AppScaleLogger.log("Reusing already running instances.")
-      # Set newly obtained node layout info for this deployment.
-
-      for i, sorted_instance_id in enumerate(sorted(instance_ids)):
-        index = instance_ids.index(sorted_instance_id)
+      # Get the node_info from the locations JSON.
+      node_info = LocalState.get_local_nodes_info(keyname=options.keyname)
+
+      previous_node_list = node_layout.from_locations_json_list(node_info)
+      # If this is None, the AppScalefile has been changed or the nodes could
+      #  not be matched up by roles/jobs.
+      if previous_node_list is None:
+        raise BadConfigurationException("AppScale does not currently support "
+                                        "changes to AppScalefile or locations "
+                                        "JSON between a down and an up. If "
+                                        "you would like to "
+                                        "change the node layout use "
+                                        "down --terminate before an up.")
+      node_layout.nodes = previous_node_list
+      for i, node in enumerate(node_layout.nodes):
```

Comment:
Please refrain from using single-letter variable names. You can use something like: _node_index_